avro4s-core/src/main/scala/com/sksamuel/avro4s/AvroInputStream.scala

```diff
@@ -95,7 +96,14 @@ object AvroInputStream {
   def json[T: SchemaFor : FromRecord](path: String): AvroJsonInputStream[T] = json(Paths.get(path))
   def json[T: SchemaFor : FromRecord](path: Path): AvroJsonInputStream[T] = json(path.toFile)
 
-  def binary[T: SchemaFor : FromRecord](in: InputStream): AvroBinaryInputStream[T] = new AvroBinaryInputStream[T](in)
+  def binary[T: SchemaFor : FromRecord](in: InputStream, writerSchema: Schema): AvroBinaryInputStream[T] = new AvroBinaryInputStream[T](in, Option(writerSchema))
```

Comment:
Ideally the writerSchema has the same schema as AvroSchema[T] as a default value on binary but we can't overload with default values :/